Makefile

```diff
@@ -100,16 +100,22 @@ prod-build:
 
 prod-build-android: export TARGET_OS ?= android
 prod-build-android:
-	lein prod-build-android
+	cp params_prod.edn params.edn && \
```

Comment:
Should this `cp` be here? Why not make it e separate target? Like:
```
prod-params:
    cp  params_prod.edn params.edn 

prod-build-android: prod-params
```
Or something like that.
Also, can you explain what this file changes exactly? Maybe you can add that as a comment to the Makefile target?

Reply:
This file changes the parameter which makes macro behave differently in prod and dev env.

Reply:
Obviously, but differently how?

Reply:
Also, who use these files? Is there any reason why this can't be managed by env variables?

Reply:
currently `defmodule` macro uses this file (introduced in this PR), another macro will need platform version in that file. This is necessary to provide env vars during cljs compile time. It can be done via env vars though might be a bit implicit. I would merge it as it is and then spend time on this later, because I'm kind of have a bit different objective. It works and I'm happy.

Reply:
Sure, just don't want to multiply entities in the repo if it's not necessary. And it doesn't have to be implicit, you can clearly define the env variable above the Makefile target and make it very clear so as to what it does, like so:
```Makefile
prod-build-android: export TARGET_OS ?= android
prod-build-android: export BUILD_ENV ?= prod
prod-build-android:
	lein prod-build-android && \
	node prepare-modules.js
```
We also have `.env*` files in the repo root which contian various settings like that, might be a good place for that too.

Reply:
for some reason `BUILD_ENV` is not seen from lein if defined this way, only if I make it `BUILD_ENV=prod lein prod-build-android && \`

Reply:
Might be because of `--pure`.
https://github.com/status-im/status-react/blob/cdae3423bcc986d9a390de0a5479cfa4c005c121/nix/shell.sh#L34-L38
For now the simplest thing to do would be to just add a `--keep BUILD_ENV` somewhere in there. What do you think @PombeirP ?
I can do a separate PR that will add an env variables to do that.

Reply:
How about this: https://github.com/status-im/status-react/pull/8398

Reply:
I've merged #8398 so try rebasing and adding an `export NIX_KEEP ?= BUILD_ENV` somewhere at the top of the `Makefile`, probably next to this:
https://github.com/status-im/status-react/blob/0e7f73fd11e396c5e5fa898178d3872eb676f234/Makefile#L29
And add a comment saying something like:
```Makefile
# Defined which variables will be kept for Nix pure shell, use semicolon as divider
```
And we should be golden.